kde-apps/okular/okular-22.04.3.ebuild

```diff
@@ -96,12 +96,47 @@ src_configure() {
 }
 
 src_test() {
-	# mainshelltest hangs, chmgeneratortest fails, bug #603116
-	# parttest hangs, bug #641728, annotationtoolbartest fails, KDE-Bug #429640
-	# epubgeneratortest and signunsignedfieldtest fail, whatever. bug #852749
+	local bad_tests=(
+		# annotationtoolbartest fails, KDE-Bug #429640
+		annotationtoolbartest
+		# mainshelltest hangs, chmgeneratortest fails, bug #603116
+		chmgeneratortest
+		mainshelltest
+		# parttest hangs, bug #641728
+		parttest
+		# epubgeneratortest and signunsignedfieldtest fail, whatever. bug #852749
+		epubgeneratortest
+		signunsignedfieldtest
+	)
+	# Many Okular tests require Okular to be installed
+	# See https://invent.kde.org/graphics/okular/-/issues/75#note_508616 and https://bugs.gentoo.org/653618#c11
+	if ! has_version kde-apps/okular; then
+		bad_tests+=(
+			addremoveannotationtest
+			annotationstest
+			annotationtoolbartest
+			calculatetexttest
+			documenttest
+			editannotationcontentstest
+			editformstest
+			epubgeneratortest
+			formattest
+			keystroketest
+			kjsfunctionstest
+			mainshelltest
+			modifyannotationpropertiestest
+			searchtest
+			signatureformtest
+			signunsignedfieldtest
+			translateannotationtest
+			visibilitytest
+		)
+		ewarn "Many tests for ${PN} are disabled because ${PN} is reqired to be installed for them to succeed."
```

Comment:
typo at `required`.

@thesamesam do we really want to go that route? imo if a package requires itself to be installed for tests, I just accept upstream's failure.

Reply:
I am not sure what upstream feel strongly about(?)

I am only concerned about maintaining this in potentially other packages as well, because I am not going to.

Reply:
They feel strongly about this not being a bug. Anyway, I'm fine with maintaining it.